backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FingerprintStepLayout.java

```diff
@@ -0,0 +1,479 @@
+package org.researchstack.backbone.ui.step.layout;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyPermanentlyInvalidatedException;
+import android.security.keystore.KeyProperties;
+import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;
+import android.support.v4.os.CancellationSignal;
+import android.support.v7.app.AlertDialog;
+import android.util.AttributeSet;
+import android.util.Base64;
+import android.view.View;
+
+import org.researchstack.backbone.R;
+import org.researchstack.backbone.StorageAccess;
+import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.step.FingerprintStep;
+import org.researchstack.backbone.step.InstructionStep;
+import org.researchstack.backbone.step.Step;
+import org.researchstack.backbone.ui.callbacks.StepCallbacks;
+import org.researchstack.backbone.utils.ResUtils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+/**
+ * Created by TheMDP on 1/31/17.
+ *
+ * This class can be used as a more convenient and secure method for creating
+ * an encryption key that will be used to protect the user's data
+ *
+ * It works with the FingerprintManager to create a secret key that is backed by the user's fingerprint
+ */
+
+@TargetApi(android.os.Build.VERSION_CODES.M) // api 23, or Android 6.0
+public class FingerprintStepLayout extends InstructionStepLayout {
```

Comment:
No, you are right.  The keystore/fingerprint/encryption code should go in some sort of security utils class.  I'll do that.